packages/name/src/formatName.ts

```diff
@@ -0,0 +1,31 @@
+import {FAMILY_NAME_FIRST_NAME_ORDERING} from './constants';
+
+// Note: A similar Ruby implementation of this function also exists at https://github.com/Shopify/shopify-i18n/blob/main/lib/shopify-i18n/name_formatter.rb.
+export function formatName({
+  name,
+  locale,
+  options,
+}: {
+  name: {givenName?: string; familyName?: string};
+  locale: string;
+  options?: {full?: boolean};
+}) {
+  if (!name.givenName) {
+    return name.familyName || '';
+  }
+  if (!name.familyName) {
+    return name.givenName;
+  }
+
+  const isFullName = Boolean(options && options.full);
+
+  const customNameFormatter = FAMILY_NAME_FIRST_NAME_ORDERING.get(locale);
```

Comment:
This is a pre-existing problem, but I just realized that this method won't return the expected results if someone happens to use this library with a Japanese or Korean locale that isn't top-level (e.g. `ja-JP` or `ko-KR`). I'll create an issue for it, something to address in a future PR!